impeller/entity/geometry/point_field_geometry.cc

```diff
@@ -135,34 +135,31 @@ GeometryResult PointFieldGeometry::GetPositionBufferGPU(
   if (radius_ < 0.0) {
     return {};
   }
-  auto determinant = entity.GetTransform().GetDeterminant();
+  Scalar determinant = entity.GetTransform().GetDeterminant();
   if (determinant == 0) {
     return {};
   }
 
   Scalar min_size = 1.0f / sqrt(std::abs(determinant));
   Scalar radius = std::max(radius_, min_size);
 
-  auto vertices_per_geom = ComputeCircleDivisions(
+  size_t vertices_per_geom = ComputeCircleDivisions(
       entity.GetTransform().GetMaxBasisLength() * radius, round_);
 
-  auto points_per_circle = 3 + (vertices_per_geom - 3) * 3;
-  auto total = points_per_circle * points_.size();
+  size_t points_per_circle = 3 + (vertices_per_geom - 3) * 3;
+  size_t total = points_per_circle * points_.size();
 
-  auto cmd_buffer = renderer.GetContext()->CreateCommandBuffer();
-  auto compute_pass = cmd_buffer->CreateComputePass();
-  auto& host_buffer = renderer.GetTransientsBuffer();
+  std::shared_ptr<CommandBuffer> cmd_buffer =
+      renderer.GetContext()->CreateCommandBuffer();
+  std::shared_ptr<ComputePass> compute_pass = cmd_buffer->CreateComputePass();
+  HostBuffer& host_buffer = renderer.GetTransientsBuffer();
 
-  auto points_data =
+  BufferView points_data =
       host_buffer.Emplace(points_.data(), points_.size() * sizeof(Point),
                           DefaultUniformAlignment());
 
-  DeviceBufferDescriptor buffer_desc;
-  buffer_desc.size = total * sizeof(Point);
-  buffer_desc.storage_mode = StorageMode::kDevicePrivate;
-
-  auto geometry_buffer = DeviceBuffer::AsBufferView(
-      renderer.GetContext()->GetResourceAllocator()->CreateBuffer(buffer_desc));
+  BufferView geometry_buffer =
+      host_buffer.Emplace(nullptr, total * sizeof(Point), alignof(Point));
```

Comment:
Emplace nullptr essentially reserves regions of the hostbuffer which can be bound as read/write for other stages. The safety of this is enforced via barrier we insert below.